File: ProjetAttractionAngular/src/app/model/compte.ts
export class Compte {

  constructor(
    protected _login?: string,
    protected _password?: string,
    protected _id?: number,
    protected _isJoueur?: boolean
  ){}


	public get login() : string|undefined {
		return this._login;
	}

	public set login(login : string|undefined) {
		this._login = login;
	}

	public get password() : string|undefined {
		return this._password;
	}

	public set password(password : string|undefined) {
		this._password = password;
	}

	public get id() : number | undefined {
    return this._id;
	}


	public set id(id : number|undefined) {
		this._id = id;
	}

	public get isJoueur() : boolean | undefined {
    return this._isJoueur;
	}


	public set isJoueur(isJoueur : boolean|undefined) {
		this._isJoueur = isJoueur;
	}

}

File: ParcAttractionBoot/src/main/java/ParcAttractionBoot/model/Compte.java
package ParcAttractionBoot.model;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;

import com.fasterxml.jackson.annotation.JsonView;


@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="type_compte",columnDefinition = "ENUM('admin', 'joueur')")
public abstract class Compte {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="id_compte")
	@JsonView(JsonViews.Common.class)
	protected Integer id;
	@JsonView(JsonViews.Common.class)
	protected String login;
	@JsonView(JsonViews.Common.class)
	protected String password;
	@Column(name="enable")
	@JsonView(JsonViews.Common.class)
	private boolean enable;
	
	@JsonView(JsonViews.Common.class)
	private transient boolean isJoueur;
	
	
	public Compte(Integer id,String login, String password) {
		this.login = login;
		this.password = password;
		this.id = id;
	}
	
	
	public Compte(String login, String password) {
		this.login = login;
		this.password = password;
	}
	

	public Compte(String login, String password,boolean isJoueur) {
		this.login = login;
		this.password = password;
		this.isJoueur = isJoueur;
	}
	
	public Compte(String login) {
		this.login = login;
		
	}
	
	
	public Compte() {
	}


	
	
	public boolean isEnable() {
		return enable;
	}


	public void setEnable(boolean enable) {
		this.enable = enable;
	}


	public Integer getId() {
		return id;
	}
	
	
	public void setId(Integer id) {
		this.id = id;
	}
	
	
	public String getLogin() {
		return login;
	}


	public void setLogin(String login) {
		this.login = login;
	}


	public String getPassword() {
		return password;
	}


	public void setPassword(String password) {
		this.password = password;
	}


	public boolean isJoueur() {
		return isJoueur;
	}


	public void setIsJoueur(boolean isJoueur) {
		this.isJoueur = isJoueur;
	}


	@Override
	public String toString() {
		return "Compte [login=" + login + ", password=" + password + "]";
	}
	

	

}
